BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -616,52 +675,27 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
     }
 }
 
-- (void)connection:(NSURLConnection *)connection didFailWithError:(NSError *)error
-{
-    BOXLog(@"Connection %@ did fail with error %@", connection, error);
-    if ([error code] != NSURLErrorUserCancelledAuthentication)
-    {
-        [self failAuthenticationWithConnection:connection challenge:nil message:[error localizedDescription] error:error];
-    }
-}
-
-#pragma mark - NSURLConnectionDataDelegate methods
-
-- (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLResponse *)response
+- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error
 {
-    BOXLog(@"Connection %@ did receive response %@", connection, response);
-    if ([response isKindOfClass:[NSHTTPURLResponse class]])
-    {
-        BOXLog(@"HTTP Headers were: %@", [(NSHTTPURLResponse *)response allHeaderFields]);
+    if (error == nil) {
+        // Success
+        BOXLog(@"URLSessionTask %@ did finish loading. Requesting that the webview load the data (%lu bytes) with reponse %@", task, (unsigned long)[self.connectionData length], self.connectionResponse);
+        [self setWebViewCanBeUsedDirectly:YES forHost:task.currentRequest.URL.host];
+        [self setWebViewCanBeUsedDirectly:YES forHost:[self.connectionResponse URL].host];
+        [(UIWebView *)self.view loadData:self.connectionData
+                                MIMEType:[self.connectionResponse MIMEType]
+                        textEncodingName:[self.connectionResponse textEncodingName]
+                                 baseURL:[self.connectionResponse URL]];
+
+        self.connectionResponse = nil;
```

Comment:
No, the auth flow has not yet completed, we're continuing the flow by loading the currently fetched data into the web view.

The completion block is only called at the end of the auth flow (due to success or errors).